### PR TITLE
Overwrite 'receivedAt' property prior to add tx to pool - Closes #629

### LINKED ISSUE #629: Node crash with the FTL error
<code>
[FTL] 2017-06-25 10:23:08 | Domain transactions - {"message":"Cannot read property 'getTime' of undefined","stack":"TypeError: Cannot read property 'getTime' of undefined\n    at /opt/install/lisk-main/app.js:12:16957\n    at /opt/install/lisk-main/node_modules/async/dist/async.js:3025:16\n    at replenish (/opt/install/lisk-main/node_modules/async/dist/async.js:882:17)\n    at /opt/install/lisk-main/node_modules/async/dist/async.js:886:9\n    at eachLimit$1 (/opt/install/lisk-main/node_modules/async/dist/async.js:3114:22)\n    at Object.<anonymous> (/opt/install/lisk-main/node_modules/async/dist/async.js:918:16)\n    at Object.l.expireTransactions (/opt/install/lisk-main/app.js:12:16828)\n    at /opt/install/lisk-main/app.js:12:13762\n    at nextTask (/opt/install/lisk-main/node_modules/async/dist/async.js:5070:14)\n    at Immediate.<anonymous> (/opt/install/lisk-main/node_modules/async/dist/async.js:5064:13)"}
</code>

A restart is needed in order to have the node up and running

### PR DESCRIPTION
**Pull request by 4miners**
Closes #629

### CODE REVIEW

**Comment by 5an1ty** [comment]
Solution is similar to what I proposed, problem correctly identified and fixed. Would it be possible to create a test for this?

**Comment by 4miners** [comment]
@5an1ty What's the point of checking `receivedAt` if we always have that property set explicitly? We don't rely on received data here.
Test suite for transaction pool would be nice, yes.

**Comment by 5an1ty** [comment]
@4miners True, you don't rely on external data here so a little bit pointless :-)!

File: logic/transactionPool.js [hunk 1]
```diff
@@ -276,9 +271,11 @@ TransactionPool.prototype.countUnconfirmed = function () {
  * @param {transaction} transaction
  */
 TransactionPool.prototype.addBundledTransaction = function (transaction) {
-	self.bundled.transactions.push(transaction);
-	var index = self.bundled.transactions.indexOf(transaction);
-	self.bundled.index[transaction.id] = index;
+	if (self.bundled.index[transaction.id] === undefined) {
+		self.bundled.transactions.push(transaction);
+		var index = self.bundled.transactions.indexOf(transaction);
```
Comment (MaciejBaj) [inline logic/transactionPool.js]: is indexOf working on array of objects here?

Comment (LucasIsasmendi) [inline logic/transactionPool.js]: Nop, it doesn't works, should be replaced by `self.bundled.transactions.findIndex(function(tr) { return tr["id"] === transaction.id })`

Comment (karmacoma) [inline logic/transactionPool.js]: We are not using `indexOf` here. Look more closely. There is `self.bundled.index` and `self.bundled.transactions`.

Comment (4miners) [inline logic/transactionPool.js]: @MaciejBaj yep, looks like it works as expected, return correct indexes when we pass an object. https://jsfiddle.net/32pwm39f/

**Pull request merged**